### PR TITLE
Add one more thing for CMake build system

### PR DESCRIPTION
**Pull request by multiplemonomials**
It turns out I missed something when I manually merged my changes to base_setup.py.  I still need to add a block that will use the environment compilers if AMBERHOME is not defined.

### CODE REVIEW

File: scripts/base_setup.py [hunk 1]
```diff
@@ -362,36 +362,42 @@ def setenv_cc_cxx(ambertools_distro, extra_compile_args, extra_link_args):
...
+                    if line.startswith('CC='):
+                        CC = line.split('=', 1)[-1]
+                        break
+
+                for line in lines:
+                    if line.startswith('CXX='):
+                        CXX = line.split('=', 1)[-1]
+                        break
+        else:
+            #detect environment variables passed from CMake script
```
Comment (hainm) [inline scripts/base_setup.py]: `#detect` -> `# detect`

**Comment by hainm** [comment]
is it standalone yet? If yes, you should add the testing here: https://github.com/Amber-MD/pytraj/blob/master/.travis.yml

**Comment by multiplemonomials** [comment]
No, it's not standalone yet.  I didn't think that there was any point in making the pytraj one standalone because it's just a CMake wrapper around your setup.py.  Do you want me to make it standalone?  I'm happy to, if you want.

**Comment by hainm** [comment]
No, it's not necessary. Just because I read your comment elsewhere

https://github.com/Amber-MD/cpptraj/pull/519#issuecomment-317220770

**Comment by hainm** [comment]
The build fails because the newest cython (0.26) is installed and it has bug. https://groups.google.com/forum/m/#!topic/cython-users/tQlwfcpdf0k

Can you force to use cython 0.25? Thanks.

**Comment by multiplemonomials** [comment]
you mean that you want me to make the CMake build system reject Cython 0.26?

**Comment by hainm** [comment]
https://github.com/Amber-MD/pytraj/blob/master/devtools/ci/setup_env.sh#L17

change to
```
conda install --yes conda-build jinja2 anaconda-client pip cython=0.25 numpy
```

**Comment by hainm** [comment]
uhm, sorry, should be this line: https://github.com/Amber-MD/pytraj/blob/43793d76d1f3b93f233d16fbe91114d626f678d0/devtools/ci/setup_env.sh#L24

**Comment by hainm** [comment]
fixed. Happy merging. thanks.

**Comment by multiplemonomials** [comment]
great!

**Pull request merged**